biz.aQute.bndlib/src/aQute/bnd/service/RepositoryPlugin.java

```diff
@@ -18,7 +19,15 @@
  * combination. It is also possible to put revisions in a repository if the
  * repository is not read only.
  */
-public interface RepositoryPlugin {
+public interface RepositoryPlugin extends Tagged {
+
+	/**
+	 * Each repo has by default the tag {@link Constants#REPOTAGS_RESOLVE} if no
+	 * tags are set at the repo definition in build.bnd That means it is
+	 * consider
+	 */
+	Tags DEFAULT_REPO_TAGS = Tags.of(Constants.REPOTAGS_RESOLVE);
```

Comment:
Hmm, this requires collaboration of the plugins. If a user implemented RepositoryPlugin he will not automatically get this default. 
The only way I know how to handle these backward compatibility cases is to treat the empty set as explicitly different from the side of the caller has a behavior for the empty set, not the callee because it has no clue how it is going to be used.

Once there is a single, even a dummy, tag, the caller can assume the user is aware of tags and is in charge. The empty set indicates the user is unaware or the plugin does not yet participate. In both case the default behavior of the _caller_ should apply.

To have a default content, you remove the possibility on the caller side to see the distinction between repos that have not (yet) been tagged and tagged. I think this is a crucial distinction?

Reply:
> If a user implemented RepositoryPlugin he will not automatically get this default.

I added a default method in `RepositoryPlugin.java` for this purpose:

https://github.com/chrisrueger/bnd/blob/add-repository-tags/biz.aQute.bndlib/src/aQute/bnd/service/RepositoryPlugin.java#L292

Maybe we should have a call about this, to get me on the same page:) My current assumption about "default-behavior" was, that all repositories by default should behave as if they have the "resolve" tag - because they should take part in the resolution. I think I interpreted the sentence somewhere... don't know where anymore. 
But anyway I am absolutely open and agree that we should first define what "default" means and then implement the right thing. 

> I think this is a crucial distinction?

Yes.  Let's stare at code together in call quick call. Today does not work for me within the next days it is possible.